### ISSUE TITLE
Dynamically create build containers and scripts

### ISSUE THREAD

**Issue by laurenzfb** [ls1intum/Artemis#7141]
# Dynamically create build containers and scripts

### Problem  
The current local CI system statically creates a Docker container for a build job and mounts a static build script on it. The container always uses the "ls1/artemis-maven-template" and the build script tests for Java exercises. This way, the local CI setup can only support Java programming exercises. Other features like auxiliary repositories, sequential test runs, static code analysis and testwise coverage are not possible because of the static build script. Being able to dynamically create containers and build scripts based on the programming language, project type and other customizations can fix this problem.

### Motivation 
The motivation seems quite obvious. We want to support multiple programming languages and project types, as well as the previously mentioned features in the programming exercises in the local CI setup. 

---
### Existing (Problematic) Solution / System 

Currently, the UI only allows you to create Java Gradle exercises. When a programming exercise is created, the template and solution repositories are created and the respective build plans created and executed. The method `runBuildJob` in `LocalCIBuildJobExecutionService.java` mounts the repositories and the static build script to the container, configures it and eventually runs it. It always uses "ls1/artemis-maven-template" for the Docker image and `build_and_run_tests.sh` as the script.

### Proposed System 

Before the build container can be configured, we need to know the programming language, the project type and other options of the programming exercise object to be able to determine which Docker image will be used and what commands will be needed for the script. Then we can create the build script, configure the container and run it.

### Requirements 
1. FR: Programming language support: Create programming exercises of other programming languages
2. FR: Project type support: Create programming exercises of other project types
3. FR: Sequential test runs: Create programming exercises with sequential test runs enabled
4. FR: Static code analysis: Create programming exercises with static code analysis enabled
5. FR: Testwise coverage: Create programming exercises with testwise coverage enabled

---
### Dynamic Behavior 

---
### Other Design Decisions

---

**Comment by Strohgelaender** [comment]
I think we should widen this topic a bit and think about not only dynamically selecting a fitting base-buildplan, but also allowing instructors to customize its content.

While not often, Instructors should be able to manually change the build script of creation exercises. Use cases for this include:

- Adding a custom script task: 
One example of this are git-commands that should be executed on the student's repository. We have exercises where students need to resolve merge conflicts, switch branches, or just generally work with some kind of automatic change to their repository. Instructrs should be able to include such additional tasks. 

- Modify execution parameters
In one exercise I was creating I had an issue with the mocking frameworks and java's (at that time new) illegal access restrictions. I was only able to fix using the `--add-opens` parameter.

- Build artifacts
On Bamboo, Instructors can specify build artifacts, that Artmis provides to the student. In most cases, this will be the student's application. Instructors should be able to specify how to build the file and where the result is located, so that it can be provided to the student.


Additionally, Instructors should be able to specify at least the version of the used docker image (e.g when you need to stick to an older version or directly need an update before the default changed in Artemis). Maybe even allow to manually select the used image to a completely different one (even though I'm not aware of cases where this is useful or needed right now)

**Comment by laurenzfb** [comment]
Yes, that is eventually the goal. I'm working towards a solution that provides a way, similar to Bamboo, to view and interact with build plans. This here is basically just a first step to ensure that Artemis is able to create dynamic build scripts. Later on, I will add the functionality for customizing the build plan. Thank you for the remarks, I will definitely take them into consideration!

**Comment by ValentinHerrmann** [comment]
By selecting the `Custom Build Configuration` option during exercise creation, the container image can be freely selected, and several other options can be configured. Therefore, this issue can be considered as resolved.

<img width="1814" height="1311" alt="Image" src="https://github.com/user-attachments/assets/e5789d8b-7c0d-42c2-883e-f0d2fff03743" />
